astroquery/mast/core.py

```diff
@@ -1394,5 +1416,417 @@ def download_products(self, products, download_dir=None,
         return manifest
 
 
+@async_to_sync
+class CatalogsClass(MastClass):
+    """
+    MAST catalog query class.
+
+    Class for querying MAST catalog data.
+    """
+
+    def __init__(self):
+
+        super(CatalogsClass, self).__init__()
+
+        self.catalogLimit = None
+
+    def _parse_result(self, response, verbose=False):
+
+        resultsTable = super(CatalogsClass, self)._parse_result(response, verbose)
+
+        if len(resultsTable) == self.catalogLimit:
+            warnings.warn("Maximum catalog results returned, may not include all sources within radius.",
+                          MaxResultsWarning)
+
+        return resultsTable
+
+    @class_or_instance
+    def query_region_async(self, coordinates, radius=0.2*u.deg, catalog="Hsc",
+                           pagesize=None, page=None, **kwargs):
+        """
+        Given a sky position and radius, returns a list of catalog entries.
+        See column documentation for specific catalogs `here <https://mast.stsci.edu/api/v0/pages.htmll>`__.
+
+        Parameters
+        ----------
+        coordinates : str or `astropy.coordinates` object
+            The target around which to search. It may be specified as a
+            string or as the appropriate `astropy.coordinates` object.
+        radius : str or `~astropy.units.Quantity` object, optional
+            Default 0.2 degrees.
+            The string must be parsable by `astropy.coordinates.Angle`. The
+            appropriate `~astropy.units.Quantity` object from
+            `astropy.units` may also be used. Defaults to 0.2 deg.
+        catalog : str, optional
+            Default HSC.
+            The catalog to be queried.
+        pagesize : int, optional
+            Default None.
+            Can be used to override the default pagesize for (set in configs) this query only.
+            E.g. when using a slow internet connection.
+        page : int, optional
+            Default None.
+            Can be used to override the default behavior of all results being returned to
+            obtain a specific page of results.
+        **kwargs
```

Comment:
might need `: dict` to get this to properly format?

Reply:
I am not sure what you mean. Do you mean in the documentation? It looks ok to me...

Reply:
OK, then it's fine (yes, I meant in the documentation)